src/Lib.hs

```diff
@@ -88,9 +120,11 @@ isValidChain chain = case chain of
 mineBlockFrom :: (MonadIO m) => Block -> String -> m Block
 mineBlockFrom lastBlock stringData = do
   time <- liftIO epoch
-  return . addHashToBlock $ Block { index        = index lastBlock + 1
-                                  , previousHash = blockHash lastBlock
-                                  , timestamp    = time
-                                  , blockData    = stringData
-                                  , blockHash    = "will be changed"
-                                  }
+  let block = Block { index        = index lastBlock + 1
+                    , previousHash = blockHash lastBlock
+                    , timestamp    = time
+                    , blockData    = stringData
+                    , nonce        = 0
+                    , blockHash    = "will be changed"
+                    }
+  return . addHashToBlock $ block { nonce = findNonce block }
```

Comment:
It looks like we manually set the nonce then add a hash in 2 places. It would be good to just pull that into a function so we don't forget to set the nonce in a new spot later on.